augur/util_support/warnings.py

```diff
@@ -0,0 +1,29 @@
+import warnings
+
+# Set up warnings & exceptions
+_deprecationWarningsEmitted = False
+
+def deprecated(message):
+    warnings.warn(message, DeprecationWarning, stacklevel=2)
+    global _deprecationWarningsEmitted
+    _deprecationWarningsEmitted=True
+
+def warn(message):
+    warnings.warn(message, UserWarning, stacklevel=2)
+
+def configure_warnings():
+    # we must only set these when someone runs `augur export v2` (i.e. run_v2() is called)
+    # else they will apply to _all_ augur commands due to the way all commands are pulled
+    # in by the augur runner (augur/__init__.py)
+    def customformatwarning(message, category, filename, lineno, line=None):
+        if category.__name__ == "UserWarning":
+            return "WARNING: {}\n\n".format(message)
+        if category.__name__ == "DeprecationWarning":
+            return "DEPRECATED: {}\n\n".format(message)
+        return "{}\n".format(message)
+
+    warnings.formatwarning = customformatwarning
+    warnings.simplefilter("default") # show DeprecationWarnings by default
```

Comment:
> Over time we will hopefully make use of these functions in other parts of Augur.

Nice! So eventually the warnings would be centrally configured and then we'd replace a lot of the `print_err("WARNING...)` calls with `warn()`.

Reply:
Yup! Looking at this after a week I should update the comment to not imply that this can only be used for export v2

Reply:
[comment updated](https://github.com/nextstrain/augur/compare/23595ee4deb4eebf66d73272f20d54dbfc420f03..b294cbcdb89cef2c2ea4ae7e5a3f8957a67abb5d)